pctasks/run/pctasks/run/workflow/kubernetes.py

```diff
@@ -0,0 +1,304 @@
+    args = task_definition.args
+
+    scaler_data = {
+        "apiVersion": "keda.sh/v1alpha1",
+        "kind": "ScaledObject",
+        "metadata": {"name": f"{prefix}-scaler"},
+        "spec": {
+            "scaleTargetRef": {"name": get_deployment_name(task_definition)},
+            "minReplicaCount": args["streaming_options"]["min_replica_count"],
+            "maxReplicaCount": args["streaming_options"]["max_replica_count"],
+            "pollingInterval": args["streaming_options"]["polling_interval"],
+            "triggers": [
+                {
+                    "type": "azure-queue",
+                    "authenticationRef": {
+                        "name": KEDA_QUEUE_CONNECTION_STRING_AUTH_NAME
+                    },
+                    "metadata": {
+                        "queueName": queue_name,
+                        "queueLength": str(
+                            args["streaming_options"]["trigger_queue_length"]
+                        ),
+                        "accountName": account_name,
+                    },
+                }
+            ],
+        },
+    }
+    return scaler_data
+
+
+def build_streaming_deployment(
+    task_definition: TaskDefinition,
+    input_uri: str,
+    taskio_tenant_id: str,
+    taskio_client_id: str,
+    taskio_client_secret: str,
+) -> V1Deployment:
+    """
+    Build, but do not submit, a Kubernetes Deployment for a streaming task.
+
+    Parameters
+    ----------
+    task_definition : TaskDefinition
+        The pctasks TaskDefinition. This should be a streaming-style task.
+    input_uri: str
+        The URI of the input arguments for this task. It will likely be a
+        ``blob://`` style URI pointing to base64-encoded arguments.
+
+    Returns
+    -------
+    deployment
+        A Kubernetes V1Deployment object. This will have a single pod with a
+        single container.
+    """
+    env = [
+        V1EnvVar(name=k, value=v)
+        for k, v in {
+            TASKIO_TENANT_ID_ENV_VAR: taskio_tenant_id,
+            TASKIO_CLIENT_ID_ENV_VAR: taskio_client_id,
+            TASKIO_CLIENT_SECRET_ENV_VAR: taskio_client_secret,
+        }.items()
+    ]
+
+    # This is really just used for setting the azureite env vars
+    # in testing. Need to set that *before* the container starts.
+    for k, v in task_definition.args.get("extra_env", {}).items():
+        env.append(V1EnvVar(name=k, value=str(v)))
+
+    container = V1Container(
+        name="run-workflow",
+        image=task_definition.image,
+        image_pull_policy="Always",
+        command=["pctasks"],
+        args=["task", "run", input_uri],
+        env=env,
+    )
+
+    queue_name = get_name_prefix(task_definition.args["streaming_options"]["queue_url"])
+
+    common_labels = {
+        "node_group": "pc-lowlatency",
+        "planetarycomputer.microsoft.com/queue_url": queue_name,
+    }
+
+    # TODO: enable node_selector. Disabled for testing in kind.
```

Comment:
I'm not sure :/ Personally, I'm eager to merge this sooner rather than later and deal with issues as they come up.

In this specific case, I don't yet know how important a separate (autoscaling) node group is for low latency, since I don't really know what kind of load to expect. Once we start doing some heavier tasks like goes-cmi I think we'll be able to get a better idea of how important that is.

Reply:
Node selector is used now.